src/plugins/borrow-plugins/plugins/aave/BorrowEngineFactory.ts

```diff
@@ -136,8 +138,17 @@ export const makeBorrowEngineFactory = (blueprint: BorrowEngineBlueprint) => {
       collaterals: [] as BorrowCollateral[],
       debts: [] as BorrowDebt[],
       loanToValue: 0,
+      isRunning: false,
       syncRatio: 0,
 
+      async startEngine() {
+        instance.isRunning = true
+        startNetworkSyncLoop()
```

Comment:
I think this entire block needs to be wrapped in `if (!instance.isRunning)` to avoid potentially starting multiple loops? Then that way the callers don't need to remember to check for `isRunning` before calling startEngine()

Reply:
Good idea